go-controller/pkg/util/net.go

```diff
@@ -21,7 +21,11 @@ var NoIPError = errors.New("no IP available")
 // NextIP returns IP incremented by 1
 func NextIP(ip net.IP) net.IP {
 	i := ipToInt(ip)
```

Comment:
IIRC, these are copies of vendored functions in https://github.com/ovn-org/ovn-kubernetes/blob/4af36a20f3bedbcf7ec1c24ae435bd7db1220a64/go-controller/vendor/github.com/containernetworking/plugins/pkg/ip/cidr.go#L23 -> I don't know why we had copies of these in such a manner but what I'd do is the following:

1. Rename the function to be `NextSloppyIP`/`intToIPSloppy` in ovnk util file and then do your hack of passing length ; this way we know that we have deviated from the vendored versions of what upstream CNI supports. Leave a comment above func as a doc string explaining this.
2. Except for the few unit tests, the only part in whole code base using this will be:
```
	for _, excludeSubnet := range excludeSubnets {
		for excludeIP := excludeSubnet.IP; excludeSubnet.Contains(excludeIP); excludeIP = util.NextIP(excludeIP) {
```
so you can perhaps call the NextIPSloppy from there... rest of the places in tests where we call `NextIP` can be changed to use the vendored CNI version.

Reply:
@tssurya excludeSubnets should be range within the subnetCidr of this associated network. If the latter does not allow leading 0's, then excludeSubnets should not allow it either.

but according to https://github.com/kubernetes/kubernetes/issues/108074, it only mentioned not allowing IPv4 addresses with leading zeros. What about IPv6? I found this bug only because of an e2e multihoming IPv6 tests which request IPv6 leading 0's cidr.

Reply:
> @tssurya excludeSubnets should be range within the subnetCidr of this associated network. If the latter does not allow leading 0's, then excludeSubnets should not allow it either.
> 
> but according to [kubernetes/kubernetes#108074](https://github.com/kubernetes/kubernetes/issues/108074), it only mentioned not allowing IPv4 addresses with leading zeros. What about IPv6? I found this bug only because of an e2e multihoming IPv6 tests which request IPv6 leading 0's cidr.

@cathy-zhou good question, seems like IPV6 is not a problem: https://github.com/aojea/funny-ip-etcd-detector#ipv6 Will need to ask @aojea on what is the stand there, 

I don't know how many users will run into this issue in their deployments around leading 0's but for now since this is being observed in the test let's do the fix you have but just maybe rename the util func to `NextSloppyIP` and use that from controller code and point unit tests to use the vendored version in CNI which is the standard implementation of that.

Reply:
the latest cni vendor version fixed this problem: https://github.com/containernetworking/plugins/blob/d27fabcd837e94dc3dc4c08c257f48f8c9539d20/pkg/ip/cidr.go

Reply:
yea saw that latest CNI fixed the problem was about to come and tell that here but saw you already did it hehe